sdk/servicebus/Azure.Messaging.ServiceBus/tests/Sender/ServiceBusMessageBatchTests.cs

```diff
@@ -104,6 +104,73 @@ public void DisposeIsDelegatedToTheTransportClient()
             Assert.That(mockBatch.DisposeInvoked, Is.True);
         }
 
+        /// <summary>
+        ///   Verifies property accessors for the <see cref="AmqpMessageBatch.TryAdd" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void TryAddRespectsTheBatchLock()
+        {
+            var mockBatch = new MockTransportBatch();
+            var batch = new ServiceBusMessageBatch(mockBatch);
+            var message = new ServiceBusMessage(new byte[] { 0x21 });
+
+            Assert.That(batch.TryAdd(new ServiceBusMessage(new byte[] { 0x21 })), Is.True, "The event should have been accepted before locking.");
+
+            batch.Lock();
+            Assert.That(() => batch.TryAdd(new ServiceBusMessage(Array.Empty<byte>())), Throws.InstanceOf<InvalidOperationException>(), "The batch should not accept events when locked.");
+
+            batch.Unlock();
+            Assert.That(batch.TryAdd(new ServiceBusMessage(Array.Empty<byte>())), Is.True, "The event should have been accepted after unlocking.");
```

Comment:
```suggestion
            Assert.That(batch.TryAdd(new ServiceBusMessage(new byte[] { 0x21 })), Is.True, "The message should have been accepted before locking.");

            batch.Lock();
            Assert.That(() => batch.TryAdd(new ServiceBusMessage(Array.Empty<byte>())), Throws.InstanceOf<InvalidOperationException>(), "The batch should not accept messages when locked.");

            batch.Unlock();
            Assert.That(batch.TryAdd(new ServiceBusMessage(Array.Empty<byte>())), Is.True, "The message should have been accepted after unlocking.");
```

Reply:
Applying these by hand; suggested changes tend to cause havok in my rebases.  :smile: